### PR TITLE
Saving result in database after using trigger all functionality

### PR DESCRIPTION
**Pull request by FlorianGlombik**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.

### Motivation and Context

Resolves https://github.com/ls1intum/Artemis/issues/3689

### Description

Fixing the linked issue https://github.com/ls1intum/Artemis/issues/3689 by saving the result instead of the submission.

### Steps for Testing

You will need a manually assessed programming exercise with at least one manually assessed participation.

1. Create a programming exercise with manual assessment
2. Participate as a student
3. Assess the programming exercise manually
4. Change the grading of the respective exercise (e.g. change the weight, add bonus points to a test, etc.)
5. Check whether the changes are displayed correctly after reloading the scores page (this may take a few minutes depending on the capacity of the CIS)

Further interesting scenarios:
- Repeat the steps for an exam programming exercise.
- Change the tests (e.g. a student could now get a build failed)

### Screenshots

The score of the semi-automatic assessment should be adapted properly after triggering changed tests.
image

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/domain/enumeration/SubmissionType.java [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package de.tum.in.www1.artemis.domain.enumeration;
 
 /**
- * The SubmissionType enumeration.
+ * MANUAL: submission triggered by a student
+ * EXTERNAL: submission that was added through the add external submission dialogue
+ * ILLEGAL: submission of an exam programming exercise that is submitted after the individual exercise end date
```
Comment (dfuchss) [inline src/main/java/de/tum/in/www1/artemis/domain/enumeration/SubmissionType.java:6]: Maybe add "or before the individual exercise start" to the javadoc for ILLEGAL :)

**Review by dfuchss — COMMENTED** [review]
Just some comment. I'll test this PR on our test systems iff needed :)

File: src/main/java/de/tum/in/www1/artemis/domain/enumeration/SubmissionType.java (hunk 1 above)
Comment (FlorianGlombik) [reply src/main/java/de/tum/in/www1/artemis/domain/enumeration/SubmissionType.java:6]: Thanks for the heads up 👍

Comment (FlorianGlombik) [reply src/main/java/de/tum/in/www1/artemis/domain/enumeration/SubmissionType.java]: Fixed in https://github.com/ls1intum/Artemis/pull/3701/commits/ae4471168560bd6d935f9b3035361869022145b2

**Comment by FlorianGlombik** [comment]
@dfuchss https://github.com/ls1intum/Artemis/pull/3701#pullrequestreview-701964520 Help for testing is always appreciated :)

**Review by katjanakosic — COMMENTED** [review]
Somehow I get an internal server error when creating a programing exercise on TS2. It also doesn't show up in the exercises list: 
Bildschirmfoto 2021-07-12 um 19 57 49

Does someone have the same issue?

**Comment by FlorianGlombik** [comment]
@katjanakosic https://github.com/ls1intum/Artemis/pull/3701#pullrequestreview-704397266 There currently is an issue with ts2 and creating programming exercises.

**Review by cremertim — COMMENTED** [review]
tested on ts1 with @PatZintel . Worked for me after pressing Re-evaluate all :)

**Review by PatZintel — APPROVED** [review]
tested on ts1, after re-evaluating all, the changes in grading were adapted correctly

**Review by cremertim — APPROVED** [review]

**Review by maximilianbergmann — APPROVED** [review]
Tested on TS3

**Pull request merged**